src/libraries/Microsoft.Extensions.Logging/src/LoggerFactoryScopeProvider.cs

```diff
@@ -48,12 +47,48 @@ void Report(Scope current)
                     }
 
                     callback(activityLogScope, state);
+
+                    // Tags and baggage are opt-in and thus we assume that most of the time it will not be used.
+                    if ((_activityTrackingOption & ActivityTrackingOptions.Tags) != 0)
+                    {
+                        const string additionalItemsTagsPropertyKey = "__ActivityTagItemsLogScope__";
+                        AdditionalActivityItemsLogScope<object?> scope = LogScopeForAdditionalActivityItems(activity, activity.TagObjects, additionalItemsTagsPropertyKey);
+                        // Only add the scope if it's not empty
+                        if (scope.GetEnumerator().MoveNext())
+                        {
+                            callback(scope, state);
+                        }
+                    }
+
+                    if ((_activityTrackingOption & ActivityTrackingOptions.Baggage) != 0)
+                    {
+                        const string additionalItemsBaggagePropertyKey = "__ActivityBaggageItemsLogScope__";
+                        AdditionalActivityItemsLogScope<string?> scope = LogScopeForAdditionalActivityItems(activity, activity.Baggage, additionalItemsBaggagePropertyKey);
+
+                        // Only add the scope if it's not empty
+                        if (scope.GetEnumerator().MoveNext())
+                        {
+                            callback(scope, state);
+                        }
+                    }
                 }
             }
 
             Report(_currentScope.Value);
         }
 
+        private static AdditionalActivityItemsLogScope<TValue> LogScopeForAdditionalActivityItems<TValue>(Activity activity, IEnumerable<KeyValuePair<string, TValue>> items, string propertyKey)
```

Comment:
I decided to implement this option because my analysis was wrong.
First I tried to overwrite the `ToString()` because it's more lightweight.
However, I realized that this does not work for baggage: 
https://github.com/dotnet/runtime/blob/79ae74f5ca5c8a6fe3a48935e85bd7374959c570/src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs#L317

Also I decided to add the instance as ActivityProperty because then I've only to create one instance (not an instance per call).